src/FileUploader.tsx

```diff
@@ -92,7 +96,7 @@ const drawDescription = (
     multiple}
  * @returns JSX Element
  */
-const FileUploader: React.FC<Props> = (props: Props): JSX.Element => {
+const FileUploader: React.FC<Props> = forwardRef<ImperativeRefType, Props>((props,ref): JSX.Element => {
```

Comment:
What about passing a prop "checkDraggingState" which is a function that is called whenever the "dragging" value change?
Here is a snippet that might five an idea of what i think:
``` 
useEffect(()=>{
    checkDraggingState?.(dragging)
  },[dragging]);
  ```
This will let us get the value without using the imperative approach.

Reply:
Also, I want to avoid using the ref prop for this case as we might need it for a future case.
Thank you so much for your contribution.

Reply:
@KarimMokhtar  I see,  It seems better getting dragging state without imperativeHandle . As you mentioned above, I removed previous refs and used useEffect. plz check